aries_cloudagent/resolver/did_resolver.py

```diff
@@ -65,6 +70,9 @@ async def dereference(
     ) -> Union[Service, VerificationMethod]:
         """Dereference a DID URL to its corresponding DID Doc object."""
         # TODO Use cached DID Docs when possible
-        did_url = DIDUrl.parse(did_url)
-        doc = await self.resolve(profile, did_url.did)
-        return doc.dereference(did_url)
+        try:
+            did_url = DIDUrl.parse(did_url)
+            doc = await self.resolve(profile, did_url.did)
+            return doc.dereference(did_url)
+        except (DIDError) as err:
+            raise ResolverError(reason=repr(err))
```

Comment:
```suggestion
        except DIDError as err:
            raise ResolverError(reason="Failed to parse DID URL from {}".format(did_url)) from err
```

Don't go crazy with repr lol

Reply:
I will manually add this for `signature's` sake.